receiver/restapireceiver/logs.go

```
@@ -0,0 +1,115 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restapireceiver
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+// convertJSONToLogs converts an array of JSON objects to plog.Logs.
+// Each JSON object becomes one log record.
+func convertJSONToLogs(data []map[string]any, logger *zap.Logger) plog.Logs {
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	for _, item := range data {
+		logRecord := scopeLogs.LogRecords().AppendEmpty()
+
+		// Set observed timestamp
+		logRecord.SetObservedTimestamp(now)
+
+		// Set body as map from JSON object
+		if err := logRecord.Body().SetEmptyMap().FromRaw(item); err != nil {
+			logger.Warn("unable to set log body", zap.Error(err))
+			// Fallback to string representation
+			logRecord.Body().SetStr("failed to parse log body")
```

In the event we fail to set the log body I think we should drop/delete this log rather than setting the log body to this error